### DISCUSSION TITLE
bad network performance intra-cluster

### DISCUSSION THREAD

**Discussion by chris93111** [k3s-io/k3s#8787] [Q&A]
HI, i have a cluster k3s 1.24 with 8 nodes and 400 pods

i have a issu with python api that use consul kv to database
when you call it I have big latency or poor performance, moving the API to another cluster(same k3s 1.24) leaving consul the performance is 10x better,
I also tried moving consul leaving the API, the performances are always degraded
which makes me understand that I have a problem with this cluster probably network

Cluster/vm don't have saturation cpu/memory/network

in tcpdump i don't see bad checksum udp or bug of flannel / iptables is legacy 
iptables v1.4.21: no command specified

the k3s check show :
(RHEL7/CentOS7: User namespaces disabled; add 'user_namespace.enable=1' to boot command line)
- swap: should be disabled
- routes: default CIDRs 10.42.0.0/16 or 10.43.0.0/16 already routed

This can cause bad performance ?
In this cluster i have metallb / knative / istio this generate many iptables rules, this can create a bad performance ?

**Environmental Info:**
K3s Version: 
k3s version v1.24.7+k3s1 (7af5b167)
go version go1.18.7

Node(s) CPU architecture, OS, and Version: 
Linux vlpsocfg04-node 3.10.0-1160.42.2.el7.x86_64 #1 SMP Tue Aug 31 20:15:00 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
Redhat 7.9

Cluster Configuration:
3 master
2 node with metallb and contour
8 node with longhorn / knative / istio / flannel ect

**Describe the bug:**
slow performance network

**Comment by chris93111** [comment]
https://github.com/flannel-io/flannel/issues/1823

**Comment by brandond** [comment]
EL7 in general is quite dated; for example the version of iptables you're using is from November of 2013. You might see if there are any differences when you use k3s with `--prefer-bunded-bin`, but in general i suspect this is not something that K3s can address.
